In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [15]:
gdp_df_raw = pd.read_csv('../data/gdp_per_capita.csv', skipfooter = 2, engine = 'python')

In [16]:
gdp_df_raw.head(10)

Country or Area  Year        Value  Value Footnotes
0     Afghanistan  2019  2065.036398              NaN
1     Afghanistan  2018  2033.779002              NaN
2     Afghanistan  2017  2058.383832              NaN
3     Afghanistan  2016  2057.062164              NaN
4     Afghanistan  2015  2068.265904              NaN
5     Afghanistan  2014  2102.385234              NaN
6     Afghanistan  2013  2116.465454              NaN
7     Afghanistan  2012  2075.491747              NaN
8     Afghanistan  2011  1904.559799              NaN
9     Afghanistan  2010  1957.029338              NaN

In [17]:
gdp_df_raw.tail()

Country or Area  Year        Value  Value Footnotes
6863        Zimbabwe  1994  3263.934978              NaN
6864        Zimbabwe  1993  3033.504852              NaN
6865        Zimbabwe  1992  3054.889178              NaN
6866        Zimbabwe  1991  3426.598094              NaN
6867        Zimbabwe  1990  3324.348171              NaN

In [19]:
gdp_df = gdp_df_raw.drop(columns = 'Value Footnotes')
gdp_df.rename(columns = {'Country or Area' : 'Country', 'Value' : 'GDP_Per_Capita'}, inplace = True)
gdp_df.head()

Country  Year  GDP_Per_Capita
0  Afghanistan  2019     2065.036398
1  Afghanistan  2018     2033.779002
2  Afghanistan  2017     2058.383832
3  Afghanistan  2016     2057.062164
4  Afghanistan  2015     2068.265904

In [26]:
print(len(gdp_df.axes[0]))

6868


In [27]:
print(len(gdp_df.axes[1]))

3


In [31]:
gdp_df.dtypes

Country            object
Year                int64
GDP_Per_Capita    float64
dtype: object

## Number 6
The DataFrame has 3 columns and 6868 rows.
Country is and object, year is and,int and GDP is a float.

In [39]:
gdp_df['Year'].unique()

array([2019, 2018, 2017, 2016, 2015, 2014, 2013, 2012, 2011, 2010, 2009,
       2008, 2007, 2006, 2005, 2004, 2003, 2002, 2001, 2000, 1999, 1998,
       1997, 1996, 1995, 1994, 1993, 1992, 1991, 1990], dtype=int64)